tornado-api/src/main/java/uk/ac/manchester/tornado/api/types/tensors/TensorFP16.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2024, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.manchester.tornado.api.types.tensors;
+
+import uk.ac.manchester.tornado.api.annotations.Parallel;
+import uk.ac.manchester.tornado.api.internal.annotations.SegmentElementSize;
+import uk.ac.manchester.tornado.api.types.HalfFloat;
+import uk.ac.manchester.tornado.api.types.arrays.HalfFloatArray;
+import uk.ac.manchester.tornado.api.types.arrays.TornadoNativeArray;
+
+import java.lang.foreign.MemorySegment;
+import java.util.Arrays;
+
+import static java.lang.foreign.ValueLayout.JAVA_SHORT;
+
+@SegmentElementSize(size = 2)
+public final class TensorFP16 extends Tensor {
+    private static final int HALF_FLOAT_BYTES = 2;
+    /**
+     * The data type of the elements contained within the tensor.
+     */
+    private final DType dType;
+    private final Shape shape;
+
+    private final HalfFloatArray tensorStorage;
+
+    /**
+     * The total number of elements in the tensor.
+     */
+    private int numberOfElements;
+
+    /**
+     * The memory segment representing the tensor data in native memory.
+     */
+
+    public TensorFP16(Shape shape) {
+        super(DType.HALF_FLOAT, shape);
```

Comment:
FP16 is 2 bytes. so it shoult be `SHORT`, correct?

Reply:
Indeed, but the DType class has the dtypes in format that are stored in model files